File: src/test/java/de/tilosp/chess/lib/PlayerColorTest.java
package de.tilosp.chess.lib;

import static org.junit.Assert.*;
import org.junit.Test;

public class PlayerColorTest {

    @Test
    public void testLength() {
        assertEquals(2, PlayerColor.values().length);
    }

    @Test
    public void testOtherColor() {
        assertEquals(PlayerColor.WHITE, PlayerColor.BLACK.otherColor());
        assertEquals(PlayerColor.BLACK, PlayerColor.WHITE.otherColor());
    }

    @Test
    public void testOrdinal() {
        assertEquals(0, PlayerColor.WHITE.ordinal());
        assertEquals(1, PlayerColor.BLACK.ordinal());
    }

    @Test
    public void testSign() {
        assertEquals(1, PlayerColor.WHITE.sign);
        assertEquals(-1, PlayerColor.BLACK.sign);
    }

    @Test
    public void testName() {
        assertEquals("WHITE", PlayerColor.WHITE.name());
        assertEquals("BLACK", PlayerColor.BLACK.name());
    }
}

File: src/test/scala/de/tilosp/chess/lib/PlayerColorTest.scala
package de.tilosp.chess.lib

import org.junit.Assert._
import org.junit.Test

class PlayerColorTest {

  @Test
  def testOtherColor() = {
    assertEquals(PlayerColor.WHITE, PlayerColor.BLACK.otherColor)
    assertEquals(PlayerColor.BLACK, PlayerColor.WHITE.otherColor)
  }

  @Test
  def testID() = {
    assertEquals(0, PlayerColor.WHITE.ordinal)
    assertEquals(1, PlayerColor.BLACK.ordinal)
  }

  @Test
  def testSign() = {
    assertEquals(1, PlayerColor.WHITE.sign)
    assertEquals(-1, PlayerColor.BLACK.sign)
  }

  @Test
  def testName() = {
    assertEquals("WHITE", PlayerColor.WHITE.name)
    assertEquals("BLACK", PlayerColor.BLACK.name)
  }
}
